CRM/Civicase/Hook/PostProcess/SaveCaseCategoryCustomFields.php

```diff
@@ -20,9 +20,10 @@ public function run($formName, CRM_Core_Form $form) {
       return;
     }
 
+    $caseCategoryCustomFields = new CaseCategoryCustomFieldsSetting();
     $caseCategoryValues = $form->getVar('_submitValues');
 
-    CaseCategoryCustomFieldsSetting::save($caseCategoryValues['value'], [
+    $caseCategoryCustomFields->save($caseCategoryValues['value'], [
```

Comment:
We need to handle the scenario when the case category gets deleted so that the custom fields setting for that gets deleted also. See: https://github.com/compucorp/uk.co.compucorp.civiawards/blob/master/CRM/CiviAwards/Hook/PostProcess/SaveFinanceManagement.php#L27-L30

Reply:
Thanks a lot @tunbola ! Some notes:

The original `CaseCategoryInstanceBase ` base class was only running on create/update and not delete. To be able to still ensure the original forms run only for create/update, but this `SaveCaseCategoryCustomFields` class runs on create/update/delete I moved part of `CaseCategoryInstanceBase`'s `shouldRun` to a new function and reimplemented `shouldRun` on `SaveCaseCategoryCustomFields` so it supports delete.